Sources/ASN1Swift/ASN1Decoder+Utils.swift

```diff
@@ -167,9 +167,18 @@ func fetchTag(from ptr: UnsafePointer<UInt8>, size: Int, to rTag: inout ASN1Tag)
 				skipped = -1
 				break
 			}
-		}else{
+		}
+		else {
 			val = (val << 7) | UInt(b)
-			rTag = UInt8(val << 2) | rawTagClass;
+
+			// Make sure it won't overflow
+			if val < 64 {
```

Comment:
So it's specifically if it's 64 or higher than shifting left 2 bits overflows.  e.g, 64 << 1 == 128, 64 << 2 == 256.  So if it's 63 or less the shift can safely happen.  We could also possibly use the truncating formats of the UInt8 case, but I'm genuinely not sure if it can parse the receipt properly if it were to lose one of the tags

Reply:
Yeah I understand why 64 is there, Uint8.max >> 2 is 63, I just wanted to say that when reading the code it's easier to understand the purpose of this if statement if we say 
if val <= Uint8.max >> 2 {
than 
if val < 64 {
The first one expresses better the fact that we need a value here that is smaller or equal than the biggest UInt8 shifeted right 2 bits.